### ISSUE TITLE
Recursive import in Hyperspy.components1d

### ISSUE THREAD

**Issue by harripj** [hyperspy/hyperspy#2066]
Hi,

I believe there is a recursive import in when importing components1d from main module. I found this when trying to use hyperspyui, and confirmed when using a basic python console.

Hyperspy/_components/power_law.py attempts to import Hyperspy._signals.BaseSignal.

In turn Hyperspy._signals imports eels, which tries to import power_law.

This raises an import error.

Fixed right now with:
from hyperspy.signal import BaseSignal
in file Hyperspy/_components/power_law.py

and remove 'signal' in line 125, 126.

**Comment by ericpre** [comment]
Yes, this is correct and this is an annoying bug, which slip in in one the last commits of the last hyperspy release... this is partially fixed by #2054 but an appropriate test needs to be added and possibly define more precisely what is part of the public API.

Using the hyperspy api works fine though.
```python
import hyperspy.api as hs
PL = hs.model.components1D.PowerLaw()
```

**Comment by ericpre** [comment]
Duplicate of #2053.

**Comment by harripj** [comment]
Thanks, I believe this is causing a crash in the latest build of hyperspyUI too.

**Comment by ericpre** [comment]
Fixed in #2054.
